### ISSUE TITLE
Questions about batchnorm_backward_layer

### ISSUE THREAD

**Issue by csjiangwm** [pjreddie/darknet#1634]
I have found your code in src/batchnorm_layer.c  void mean_delta_cpu(...)  computes l.mean_delta by calculating dloss/dx_hat * dx_hat/dmean, and void normalize_delta_cpu(...) updates l.delta by calculating delta[index] * 1./(sqrt(variance[f] + .00001f)) + variance_delta[f] * 2. * (x[index] - mean[f]) / (spatial * batch) + mean_delta[f]/(spatial*batch), but the reference of Batch normalization computes l.mean_delta by calculating dloss/dx_hat * dx_hat/dmean + dloss/dvar*dvar/dmean, and updates l.delta by calculating delta[index] * 1./(sqrt(variance[f] + .00001f)) + variance_delta[f] * 2. * (x[index] - mean[f]) / batch + mean_delta[f]/batch. I have tried your way for training and testing, it works well. I'm so confused about it.
